lib/ObjWriter/objwriter.cpp

```diff
@@ -550,22 +547,119 @@ extern "C" void EmitDebugFileInfo(ObjectWriter *OW, int FileId,
   OST.EmitCVFileDirective(FileId, FileName);
 }
 
-// This should be invoked at the end of function code emission to flush
-// debug function info.
-extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
-                                      const char *FunctionName,
-                                      int FunctionSize) {
+static void EmitPDBDebugVarInfo(MCObjectStreamer &OST, const MCSymbol *Fn,
+                                DebugVarInfo LocInfos[], int NumVarInfos) {
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+  assert(MOFI->getObjectFileType() == MOFI->IsCOFF);
+
+  for (int i = 0; i < NumVarInfos; i++) {
+    // Emit an S_LOCAL record
+    DebugVarInfo var = LocInfos[i];
+    LocalSym sym;
+
+    memset(&sym, 0, sizeof(sym));
+
+    int sizeofSym = sizeof(LocalSym);
+    short rectyp = SymbolRecordKind::S_LOCAL;
+    short reclen = sizeof(rectyp) + sizeofSym + var.name.length() + 1;
```

Comment:
Using SymRecord struct is fine, but still need to explicitly set the fields everywhere, like
SymRecord rec = {sizeof(rectyp) + sizeofSym + var.name.length() + 1, SymbolRecordKind::S_LOCAL}